### ISSUE TITLE
v1/v2 Select Popover does not work properly

### ISSUE THREAD

**Issue by JuliusRuppert** [DominoKit/domino-ui#923]
**Describe the bug**
Searchable select from [Domino-UI Demo](https://dominokit.com/domino-ui/demo/v1/forms/basic-form-elements) adds Popover clickhandler to the whole select. So if you click on the part of the select which is not the button, the popup opens -- which I think is not intended.

**To Reproduce**
1. Go to Searchable select from [Domino-UI Demo](https://dominokit.com/domino-ui/demo/v1/forms/basic-form-elements)
2. Click on the search field and enter some text
3. Click on the plus button
4. Click aywhere next to the Popover or hit the escape button
5. Click on the part of the select which is not the button
6. The popover opens

**Expected behavior**
The `Popover` should only be opened when hitting the plus button. Unfortunately it is only possible to create a Popover that opens when clicking the element is is pointing to since it is defined in the constructor.

What I have achieved so far:
The `Popover` has a `discard` method which will remove the clickhandler on the targetelement and can be called e.g. when a button is clicked. But when the Popover is closed by hitting the escape key or clicking next to the Popover, `remove()` is called and since there is no option to add a removeHandler to the Popover, it is not possible to call `discard` and remove the clickhandler of the select element.

**Additional context**
If possible, it would be great if youu could add the fix to v1 as well.

**Comment by vegegoku** [comment]
Ok so just to get things right, this popup is not a part of the core domino-ui library itself but an implementation sample in the demo app, this means the implementation is open for the user to add what he needs to handle missing items. with that said the fix in the demo implementation should be simple, if you are copying the code from there just make sure to call `popover.detach` after calling `popover.close`

```java
.setMissingItemHandler((select, token, onComplete) -> {
                                            TextBox countryName = TextBox.create("Name")
                                                    .setRequired(true)
                                                    .withValue(token);
                                            TextBox countryCode = TextBox.create("code")
                                                    .setRequired(true);
                                            Popover.create(select.getWrapperElement())
                                                    .addCss(dui_rounded_sm)
                                                    .setPosition(DropDirection.BEST_MIDDLE_UP_DOWN)
                                                    .apply(popover -> {
                                                        popover.appendChild(Card.create("Create country")
                                                                .setIcon(Icons.map_marker())
                                                                .addCss(dui_elevation_0, dui_m_2px, dui_rounded_sm)
                                                                .appendChild(countryName)
                                                                .appendChild(countryCode)
                                                                .appendChild(div().addCss(dui_flex, dui_justify_center, dui_p_1)
                                                                        .appendChild(Button.create("Create country").addClickListener(evt -> {
                                                                            if (countryName.validate().isValid() && countryCode.validate().isValid()) {
                                                                                SelectOption<String> option = SelectOption.create(countryCode.getValue().toUpperCase(), countryCode.getValue().toUpperCase(), countryName.getValue());
                                                                                onComplete.accept(option);
                                                                                select.selectOption(option);
                                                                                popover.close();
                                                                                popover.detach();
                                                                            }
                                                                        }))
                                                                )
                                                        );
                                                    })
                                                    .addExpandListener(() -> {
                                                        select.getOptionsMenu().close();
                                                        countryName.focus();
                                                    })
                                                    .open();

                                        })
```
